specs/tabs/tabs.md

```diff
@@ -0,0 +1,173 @@
+            Tab Two
+        </button>
+        <button class="tab" part="tab" role="tab" aria-selected="false" aria-controls="panel-3" id="tab-3" tabindex="-1">
+            Tab Three
+        </button>
+        <slot name="after-content" part="after-content"></slot>
+    </div>
+    <div class="tab-panel" part="tab-panel" id="panel-1" role="tabpanel" tabindex="0" aria-labelledby="tab-1">
+        Content of the first tab
+    </div>
+    <div class="tab-panel" part="tab-panel" id="panel-2" role="tabpanel" tabindex="0" aria-labelledby="tab-2" hidden>
+        Content of the second tab
+    </div>
+    <div class="tab-panel" part="tab-panel" id="panel-3" role="tabpanel" tabindex="0" aria-labelledby="tab-3" hidden>
+        Content of the third tab
+    </div>
+</div>
+```
+
+## Implementation
+
+*Logical grouping option 1:*
+```HTML
+<fast-tabs>
+    <fast-tab-pane tab="Tab One" key="1">
+        Content of the first tab
+    </fast-tab-pane>
+    <fast-tab-pane tab="Tab Two" key="2">
+        Content of the second tab
+    </fast-tab-pane>
+    <fast-tab-pane tab="Tab Three" key="3">
+        Content of the third tab
+    </fast-tab-pane>
+    <div slot="before-content">
+        Before content
+    </div>
+    <div slot="after-content">
+        Before content
+    </div>
+</fast-tabs>
+```
+
+*Logical grouping option 2:*
+```HTML
+<fast-tabs>
+    <fast-tab-item>
+        <fast-tab>Tab One</fast-tab>
+        <fast-tab-panel>Content of the first tab</fast-tab-panel>
+    </fast-tab-item>
+    <fast-tab-item>
+        <fast-tab>Tab Two</fast-tab>
+        <fast-tab-panel>Content of the Second tab</fast-tab-panel>
+    </fast-tab-item>
+    <fast-tab-item>
+        <fast-tab>Tab Three</fast-tab>
+        <fast-tab-panel>Content of the Third tab</fast-tab-panel>
+    </fast-tab-item>
+    <div slot="before-content">
+        Before content
+    </div>
+    <div slot="after-content">
+        Before content
+    </div>
+</fast-tabs>
+```
+
+*DOM grouping:*
+```HTML
+<fast-tabs>
+    <fast-tab-list>
+        <fast-tab>Tab One</fast-tab>
+        <fast-tab>Tab Two</fast-tab>
+        <fast-tab>Tab Three</fast-tab>
+    </fast-tab-list>
+    <fast-tab-panel>Content of the first tab</fast-tab-panel>
+    <fast-tab-panel>Content of the Second tab</fast-tab-panel>
+    <fast-tab-panel>Content of the Third tab</fast-tab-panel>
+    <div slot="before-content">
+        Before content
+    </div>
+    <div slot="after-content">
+        Before content
+    </div>
+</fast-tabs>
+```
+
+### States
+
+*Tabs* can either be controlled or uncontrolled, meaning if `activeId` is passed the app author is taking control of the selected tab. The `onChange` event fires differently depending on the `tab-activation` attribute, on tab focus if `tab-activation` is set to `"focus"` or on mouse click or "spacebar" press if `tab-activation` is set to `"key-press"`.
```

Comment:
Can you provide a code-sample on how this `activeId` would be used? I'm struggling to visualize how an end-user would use this and where the IDs come from.

Reply:
An app author could set the `activeId` on the tabs.

```
<fast-tabs activeId="tab2">
    <fast-tab-item id="tab1">
        <fast-tab>Tab One</fast-tab>
        <fast-tab-panel>Content of the first tab</fast-tab-panel>
    </fast-tab-item>
    <fast-tab-item id="tab2">
        <fast-tab>Tab Two</fast-tab>
        <fast-tab-panel>Content of the Second tab</fast-tab-panel>
    </fast-tab-item>
    <fast-tab-item id="tab3">
        <fast-tab>Tab Three</fast-tab>
        <fast-tab-panel>Content of the Third tab</fast-tab-panel>
    </fast-tab-item>
</fast-tabs>
```

Reply:
If the customer is setting the component up in JS, this allows them to create the tab items and set the active item without needing to set up an id. This also enables templating libraries that support the "ref" concept to let their users assign the active tab using a ref.

So, those are a couple of use cases. Beyond that, for some of our other web components, we've been following a pattern of using a control ref property with a corresponding id property. So, I'd like to do this for consistency both with that and the native platform, which uses the same pattern for controls like label.